db/postgres-client.py

```diff
@@ -0,0 +1,258 @@
+
+def __verify_connect_to_db(conn):
+    cursor = conn.cursor()
+    cursor.execute('select * from tagstate')
+    row = cursor.fetchone()  
+    print()
+    while row:  
+        print(str(row[0]) + " " + str(row[1]))    
+        row = cursor.fetchone()
+
+def id_generator(size=6, chars=string.ascii_uppercase + string.digits):
+    return ''.join(random.choice(chars) for _ in range(size))
+
+def generate_test_image_infos(count):
+    list_of_image_infos = []
+    for i in range(count):
+        file_name = f'{id_generator(size=random.randint(4,10))}.jpg'
+        image_url = f'https://mock-storage.blob.core.windows.net/new-uploads/{file_name}'
+        img = ImageInfo(file_name,image_url,random.randint(100,600),random.randint(100,600))
+        list_of_image_infos.append(img)
+    return list_of_image_infos
+
+def get_image_ids_for_new_images(conn, list_of_image_infos):
+    url_to_image_id_map = {}
+    if(len(list_of_image_infos) > 0):
+        cursor = conn.cursor()
+        for img in list(list_of_image_infos):
+            query = "INSERT INTO Image_Info (OriginalImageName,ImageLocation,Height,Width) VALUES ('{0}','{1}',{2},{3}) RETURNING ImageId;"
+            cursor.execute(query.format(img.image_name,img.image_url,str(img.height),str(img.width)))
+            new_img_id = cursor.fetchone()[0]
+            url_to_image_id_map[img.image_url] = new_img_id
+            #__update_image(conn,[new_img_id],ImageTagState.NOT_READY)
+        conn.commit()
+    print(f"Inserted {len(url_to_image_id_map)} images to the DB")
+    return url_to_image_id_map
+
+def get_new_images(conn,number_of_images):
+    cursor = conn.cursor()
+
+    # GET N existing UNTAGGED rows
+    selected_images_to_tag = {}
+    query = ("SELECT b.ImageId, b.ImageLocation, a.TagStateId FROM Image_Tagging_State a "
+            "JOIN Image_Info b ON a.ImageId = b.ImageId WHERE a.TagStateId = 1 order by "
+            "a.createddtim DESC limit {0}")
+    cursor.execute(query.format(number_of_images))
+    for row in cursor:
+        print('Image Id: {0} \t\tImage Name: {1} \t\tTag State: {2}'.format(row[0], row[1], row[2]))
+        selected_images_to_tag[str(row[0])] = str(row[1])
+
+    __update_image(conn,selected_images_to_tag,ImageTagState.TAG_IN_PROGRESS)
+    return selected_images_to_tag.values()
+
+def update_image_urls(conn,image_id_to_url_map):
+     for image_id, new_url in image_id_to_url_map.items():
+        cursor = conn.cursor()
+        query = "UPDATE Image_Info SET ImageLocation = '{0}', ModifiedDtim = now() WHERE ImageId = {1}"
+        cursor.execute(query.format(new_url,image_id))
+        print(f"Updated ImageId: {image_id} to new ImageLocation: {new_url}")
+        __update_image(conn,[image_id],ImageTagState.READY_TO_TAG)
+        print(f"ImageId: {image_id} to has a new state: {ImageTagState.READY_TO_TAG.name}")
+        conn.commit()
```

Comment:
Error handling/rollback in the case of transaction failure for some reason?

Reply:
The conn.commit() on line 91 commits the transaction. I agree more testing is needed for cause like that but it's nothing we can't recover from. I can create a issue. See:  https://pythonhosted.org/pg8000/dbapi.html#pg8000.Connection.commit